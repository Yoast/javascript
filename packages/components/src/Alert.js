/* External dependencies */
import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import Cookies from "js-cookie";
import { __ } from "@wordpress/i18n";

/* Yoast dependencies */
import { colors } from "@yoast/style-guide";

/**
 * Returns an error icon SVG data URI.
 *
 * @param {string} color The desired color for the SVG.
 *
 * @returns {string} The SVG image data URI.
 */
const errorIcon = ( color ) =>
	"data:image/svg+xml;charset=utf8," + encodeURIComponent(
		'<svg width="1792" height="1792" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">' +
	// eslint-disable-next-line
	'<path fill="' + color + '" d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z" />' +
	"</svg>"
	);

/**
 * Returns an info icon SVG data URI.
 *
 * @param {string} color The desired color for the SVG.
 *
 * @returns {string} The SVG image data URI.
 */
export const infoIcon = ( color ) =>
	"data:image/svg+xml;charset=utf8," + encodeURIComponent(
		'<svg width="1792" height="1792" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">' +
	// eslint-disable-next-line
	'<path fill="' + color + '" d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z" />' +
	"</svg>"
	);

/**
 * Returns a warning icon SVG data URI.
 *
 * @param {string} color The desired color for the SVG.
 *
 * @returns {string} The SVG image data URI.
 */
export const warningIcon = ( color ) =>
	"data:image/svg+xml;charset=utf8," + encodeURIComponent(
		'<svg width="1792" height="1792" viewBox="0 0 576 512" xmlns="http://www.w3.org/2000/svg">' +
	// eslint-disable-next-line
	'<path fill="' + color + '" d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z" />' +	"</svg>"
	);

/**
 * Returns a success icon SVG data URI.
 *
 * @param {string} color The desired color for the SVG.
 *
 * @returns {string} The SVG image data URI.
 */
export const successIcon = ( color ) =>
	"data:image/svg+xml;charset=utf8," + encodeURIComponent(
		'<svg width="1792" height="1792" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">' +
	// eslint-disable-next-line
	'<path fill="' + color + '" d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z" />' +
	"</svg>"
	);

/**
 * Returns a close icon SVG data URI.
 *
 * @param {string} color The desired color for the SVG.
 *
 * @returns {string} The SVG image data URI.
 */
export const closeIcon = ( color ) =>
	"data:image/svg+xml;charset=utf8," + encodeURIComponent(
		'<svg width="1792" height="1792" viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg">' +
	// eslint-disable-next-line
	'<path fill="' + color + '" d="M1490 1322q0 40-28 68l-136 136q-28 28-68 28t-68-28l-294-294-294 294q-28 28-68 28t-68-28l-136-136q-28-28-28-68t28-68l294-294-294-294q-28-28-28-68t28-68l136-136q28-28 68-28t68 28l294 294 294-294q28-28 68-28t68 28l136 136q28 28 28 68t-28 68l-294 294 294 294q28 28 28 68z" />' + "</svg>"
	);

const AlertBody = styled.div`
	display: flex;
	align-items: center;
	font-size: 14px;
	border: 1px solid rgba(0, 0, 0, 0.2);
	padding: 16px;
	color: ${ props => props.alertColor };
	background: ${ props => props.alertBackground };
	margin-bottom: 20px;

	a {
		color: ${ colors.$color_alert_blue_link };

		&:hover,
		&:focus {
			color: ${ colors.$color_alert_blue_link };
		}
	}
`;

const AlertIcon = styled.span`
	align-self: flex-start;
	background-position: 50%;
	background-repeat: no-repeat;
	background-size: 1rem;
	background-image: url( ${ props => props.icon } );
	height: 1rem;
	min-width: 1rem;
	margin: 0.125rem 1em 0 0.125rem;
`;

const AlertMessage = styled.div`
	flex: 1 1 auto;

	& p:first-child {
		margin-top: 0;
	}

	& p:last-child {
		margin-bottom: 0;
	}
`;

const StyledCloseButtonTopRight = styled.button`
	align-self: flex-start;
	height: 1.25rem;
	min-width: 1.25rem;
	margin: 0 0 0 1em;
	padding: 0;
	border: 0;
	cursor: pointer;
	background: transparent url( ${ props => props.iconSource } ) no-repeat center;
	background-size: ${ props => props.iconSize };

	&:hover,
	&:focus {
		opacity: 1;
	}
`;

/**
 * Returns the rendered Alert component.
 *
 * @param {Object} props The props to use.
 *
 * @returns {ReactElement} The rendered Alert component.
 */
class Alert extends React.Component {
	/**
	 * Initializes the class with the specified props.
	 *
	 * @param {Object} props The props to be passed to the class that was extended from.
	 *
	 * @returns {void}
	 */
	constructor( props ) {
		super( props );
		this.state = {
			hideAlert: false,
		};
		this.onCrossClick = this.onCrossClick.bind( this );
		this.options = this.getTypeDisplayOptions();
	}

	/**
	 * On init, checks for the cookie name provided in the props and if present hides the alert.
	 *
	 * @returns {void}
	 */
	componentDidMount() {
		if ( Cookies.get( this.props.cookieName ) ) {
			this.setState( { hideAlert: true } );
		}
	}

	/**
	 * Returns the colors and icon to be used based on the type provided to the props.
	 *
	 * @returns {object} Options with colors and icons to be used.
	 */
	getTypeDisplayOptions() {
		switch ( this.props.type ) {
			case "error":
				return {
					color: colors.$color_alert_error_red,
					background: colors.$color_alert_error_red_light,
					icon: errorIcon( colors.$color_alert_error_red ),
					closeIcon: closeIcon( colors.$color_alert_error_red ),
				};
			case "success":
				return {
					color: colors.$color_alert_success_green,
					background: colors.$color_alert_success_green_light,
					icon: successIcon( colors.$color_alert_success_green ),
					closeIcon: closeIcon( colors.$color_alert_success_green ),
				};
			case "warning":
				return {
					color: colors.$color_alert_warning_yellow,
					background: colors.$color_alert_warning_yellow_light,
					icon: warningIcon( colors.$color_alert_warning_yellow ),
					closeIcon: closeIcon( colors.$color_alert_warning_yellow ),
				};
			case "info":
			default:
				return {
					color: colors.$color_alert_info_blue,
					background: colors.$color_alert_info_blue_light,
					icon: infoIcon( colors.$color_alert_info_blue ),
					closeIcon: closeIcon( colors.$color_alert_info_blue ),
				};
		}
	}

	/**
	 * Called on cross click, hides the alert and saves this setting in a cookie.
	 *
	 * @returns {void}
	 */
	onCrossClick() {
		Cookies.set( this.props.cookieName, "hide", { expires: 7 } );
		this.setState( { hideAlert: true } );
	}

	/**
	 * Renders the component.
	 *
	 * @returns {ReactElement} The rendered component.
	 */
	render() {
		return (
			! this.state.hideAlert &&
				<AlertBody alertColor={ this.options.color } alertBackground={ this.options.background }>
					<AlertIcon icon={ this.options.icon } />
					<AlertMessage>
						{ this.props.children }
					</AlertMessage>
					{ this.props.dismissable &&
						<StyledCloseButtonTopRight
							onClick={ this.onCrossClick }
							iconSource={ this.options.closeIcon }
							iconSize="1rem"
							aria-label={ __( "close", "yoast-components" ) }
						/>
					}
				</AlertBody>
		);
	}
}

Alert.propTypes = {
	children: PropTypes.any.isRequired,
	dismissable: PropTypes.any.isRequired,
	cookieName: PropTypes.string,
	type: PropTypes.string.isRequired,
};

Alert.defaultProps = {
	cookieName: "",
};

export default Alert;
