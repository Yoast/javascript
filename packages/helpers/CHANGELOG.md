# Change Log

All notable changes to this project will be documented in this file. Releases without a changelog entry contain only minor changes that are irrelevant for users of this library.

We follow [Semantic Versioning](http://semver.org/).

## 0.12.0 May 11th, 2020
### Added
* Adds a file that can be used to modify hidden inputs.

## 0.3.0 May 27th, 2019

### Added
* Adds `createSvgIconComponent` function to create a `SvgIcon` component with custom icons.

## 0.2.0 May 14th, 2019

### Changed
* Removes the `createComponentWithIntl` component.

## 0.1.0 April 29th, 2019

### Added
* Splits out `@yoast/helpers` from `yoast-components`. The helpers package contains helper functions required for multiple other `@yoast` packages.
* Improved handling of the `rel` attribute for links that open in a new browser's tab.
