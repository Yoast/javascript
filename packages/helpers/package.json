{
  "name": "@yoast/helpers",
  "version": "0.11.1",
  "description": "All Yoast helper functions.",
  "main": "src/index.js",
  "repository": {
    "type": "git",
    "url": "https://github.com/Yoast/javascript.git",
    "directory": "packages/helpers"
  },
  "scripts": {
    "test": "jest",
    "lint": "eslint .",
    "prepublishOnly": "rm -rf dist && cp -R src dist && cp package.json dist/package.json && json -I -f dist/package.json -e \"this.main='index.js'\" && cp .babelrc dist/.babelrc"
  },
  "author": "Yoast",
  "license": "GPL-3.0",
  "private": false,
  "dependencies": {
    "@wordpress/i18n": "^1.2.3",
    "prop-types": "^15.7.2",
    "styled-components": "^2.4.1",
    "whatwg-fetch": "1.1.1"
  },
  "peerDependencies": {
    "react": "^16.2.0"
  },
  "jest": {
    "testRegex": ".*Test.js$",
    "testEnvironment": "node",
    "setupTestFrameworkScriptFile": "<rootDir>/tools/jest/setupTests.js"
  },
  "devDependencies": {
    "@yoast/browserslist-config": "^1.2.1",
    "browserslist": "^4.7.3",
    "jest-styled-components": "^5.0.1",
    "react-test-renderer": "^16.8.6",
    "wicked-good-xpath": "^1.3.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
