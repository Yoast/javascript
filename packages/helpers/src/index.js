import { getDirectionalStyle } from "./styled-components";
import sendRequest from "./ajaxHelper";
import decodeHTML from "./htmlDecoder";
import getFeed from "./getFeed";
import getCourseFeed from "./getCourseFeed";
import getPostFeed from "./getPostFeed";
import createSvgIconComponent from "./createSvgIconComponent";

export {
	createSvgIconComponent,
	getDirectionalStyle,
	sendRequest,
	decodeHTML,
	getPostFeed,
	getCourseFeed,
	getFeed,
};

export { makeOutboundLink } from "./makeOutboundLink";
export { default as validateFacebookImage } from "./social-preview-image-validation/facebookValidation";
export { default as validateTwitterImage } from "./social-preview-image-validation/twitterValidation";
export * from "./hiddenInputHelper";
