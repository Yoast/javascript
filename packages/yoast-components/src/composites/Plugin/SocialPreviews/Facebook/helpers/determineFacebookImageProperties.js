export const SQUARE_WIDTH = 158;
export const SQUARE_HEIGHT = 158;
export const PORTRAIT_WIDTH = 158;
export const PORTRAIT_HEIGHT = 236;
export const LANDSCAPE_WIDTH = 500;
export const LANDSCAPE_HEIGHT = 261;

/**
 * Determines the image display mode of the Facebook Image.
 *
 * @param {Object} dimensions The image's dimensions.
 *
 * @returns {string} The display mode of the image.
 */
export function determineFacebookImageMode( dimensions ) {
	if ( dimensions.height > dimensions.width ) {
		return "portrait";
	}

	if ( dimensions.height === dimensions.width ) {
		return "square";
	}

	return "landscape";
}

/**
 * Gets the dimensions of the uploaded image.
 *
 * @param {string} src The image source.
 *
 * @returns {Object} The image original image dimensions.
 */
export function getOriginalImageDimensions( src ) {
	/* istanbul ignore next */
	return new Promise( ( resolve, reject ) => {
		const img = new Image();

		img.onload = () => {
			resolve( {
				width: img.width,
				height: img.height,
			} );
		};

		img.onerror = reject;

		img.src = src;
	} );
}

/**
 * Gets the ratios of the width and height of the original image in relation to the width and height
 * of the expected image.
 *
 * When we're in portrait or landscape mode, we can't just resize to the dimensions expected by Facebook. If we'd do
 * so, we would end up with warped images. That's why we calculate the ratio between the original width and height and
 * the width and height that is expected by Facebook. For example: the original image is 1600x898 and Facebook expects
 * 500x261. The width ratio would be 3.44 and the height ratio would be 3.2.
 *
 * @param {Object} dimensions The dimensions of the original image.
 * @param {string} imageMode The image mode: either portrait or landscape.
 *
 * @returns {Object} The image's width ratio and height ratio.
 */
export function getImageRatios( dimensions, imageMode ) {
	if ( imageMode === "portrait" ) {
		return {
			widthRatio: dimensions.width / PORTRAIT_WIDTH,
			heightRatio: dimensions.height / PORTRAIT_HEIGHT,
		};
	}
	if ( imageMode === "landscape" ) {
		return {
			widthRatio: dimensions.width / LANDSCAPE_WIDTH,
			heightRatio: dimensions.height / LANDSCAPE_HEIGHT,
		};
	}
}

/**
 * Gets the image dimensions that the image should have as Facebook image.
 *
 * To use as much as the allowed space as possible, we base both dimensions on the dimension with the lowest imageRatio
 * (see above). For example: for a 1600x898 image, the height ratio is larger than the width ratio. The result of dividing
 * by the heightRatio is an 1600x280.625 image. The excess of 280.625-261 = 18.375 pixels will be 'cut off' by the
 * container in the presentation part.
 *
 * If we would divide by the widthRatio, the image would become 465x261, which would mean it would not be wide enough for
 * the container which means there would be a 500-465=35px white border on one of the sides.
 *
 * @param {Object} dimensions  The dimensions of the original image.
 * @param {Object} imageRatios The ratios of the width and height of the original image in relation to the width and
 *                             height of the expected image.
 *
 * @returns {Object}           The width and height that the image should have as Facebook image.
 */
export function getImageDimensionsForFacebookImage( dimensions, imageRatios ) {
	if ( imageRatios.widthRatio <= imageRatios.heightRatio ) {
		return {
			width: dimensions.width / imageRatios.widthRatio,
			height: dimensions.height / imageRatios.widthRatio,
		};
	}

	return {
		width: dimensions.width / imageRatios.heightRatio,
		height: dimensions.height / imageRatios.heightRatio,
	};
}

/**
 * Calculates the dimensions of the image to use as FacebookImage.
 *
 * @param {Object} originalDimensions The width and height of the original image.
 * @param {string} imageMode The image mode: square, portrait or landscape.
 *
 * @returns {object} The image dimensions for the Facebook image.
 */
export function calculateFacebookImageDimensions( originalDimensions, imageMode ) {
	// Images that are too small should not be scaled.
	if ( originalDimensions.width < 158 || originalDimensions.height < 158 ) {
		return {
			width: originalDimensions.width,
			height: originalDimensions.height,
		};
	}

	/*
	 * If the image is a square, just use the squareWidth and squareHeight. We
	 * don't have to fear that the resulting image will be warped.
	 */
	if ( imageMode === "square" ) {
		return {
			width: SQUARE_WIDTH,
			height: SQUARE_HEIGHT,
		};
	}

	const imageRatios = getImageRatios( originalDimensions, imageMode );

	return getImageDimensionsForFacebookImage( originalDimensions, imageRatios );
}

/**
 * Determines the properties of the Facebook Image.
 *
 * @param {string} src The source of the image.
 *
 * @returns {Promise} The promise of the imageProperties.
 */
export function determineFacebookImageProperties( src ) {
	return getOriginalImageDimensions( src ).then( ( originalDimensions ) => {
		// Determine what image mode should be used based on the image dimensions.
		const imageMode = determineFacebookImageMode( originalDimensions );

		// Calculate the image dimensions for the specific image.
		const facebookImageDimensions = calculateFacebookImageDimensions( originalDimensions, imageMode );
		return {
			mode: imageMode,
			height: facebookImageDimensions.height,
			width: facebookImageDimensions.width,
		};
	} );
}
