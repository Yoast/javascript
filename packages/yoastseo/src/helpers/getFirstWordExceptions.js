import firstWordExceptionsEnglish from "../researches/english/firstWordExceptions.js";
import firstWordExceptionsGerman from "../researches/german/firstWordExceptions.js";
import firstWordExceptionsSpanish from "../researches/spanish/firstWordExceptions.js";
import firstWordExceptionsFrench from "../researches/french/firstWordExceptions.js";
import firstWordExceptionsDutch from "../researches/dutch/firstWordExceptions.js";
import firstWordExceptionsItalian from "../researches/italian/firstWordExceptions.js";
import firstWordExceptionsRussian from "../researches/russian/firstWordExceptions.js";
import firstWordExceptionsPolish from "../researches/polish/firstWordExceptions.js";
import firstWordExceptionsSwedish from "../researches/swedish/firstWordExceptions.js";
import firstWordExceptionsPortuguese from "../researches/portuguese/firstWordExceptions.js";
import getLanguage from "./getLanguage.js";

/**
 * Returns the first word exceptions function for a locale.
 *
 * @param {string} locale The locale to return word exceptions for.
 *
 * @returns {Function} A function that will return the first word exceptions.
 */
export default function( locale ) {
	const firstWordExceptions = {
		en: firstWordExceptionsEnglish,
		de: firstWordExceptionsGerman,
		fr: firstWordExceptionsFrench,
		es: firstWordExceptionsSpanish,
		nl: firstWordExceptionsDutch,
		it: firstWordExceptionsItalian,
		ru: firstWordExceptionsRussian,
		pl: firstWordExceptionsPolish,
		sv: firstWordExceptionsSwedish,
		pt: firstWordExceptionsPortuguese,
	};

	// If available, return the language-specific first word exceptions.
	if ( Object.keys( firstWordExceptions ).includes( getLanguage( locale ) ) ) {
		return firstWordExceptions[ getLanguage( locale ) ];
	}

	// Return the English first word exceptions as a default.
	return firstWordExceptionsEnglish;
}
