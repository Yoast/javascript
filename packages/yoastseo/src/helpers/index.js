import scoreToRating from "../interpreters/scoreToRating";
import { measureTextLength } from "./createMeasurementElement";
import getLanguagesWithWordFormSupport from "./getLanguagesWithWordFormSupport";

export {
	measureTextLength,
	scoreToRating,
	getLanguagesWithWordFormSupport,
};

export * from "./inRange";
