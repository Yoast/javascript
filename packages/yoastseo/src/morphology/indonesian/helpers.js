import { buildOneFormFromRegex } from "../morphoHelpers/buildFormRule";
import createRulesFromMorphologyData from "../morphoHelpers/createRulesFromMorphologyData";

const vowelCharacters = [ "a", "e", "i", "o", "u" ];

/**
 * Determines if an input character is a vowel.
 *
 * @param {string} character The character to check.
 *
 * @returns {boolean} Whether the input character is an Indonesian vowel.
 */
function isVowel( character ) {
	return vowelCharacters.includes( character );
}

/**
 * Calculates the total number of syllables in the input word.
 *
 * @param {string} word The word to calculate the number of syllables in.
 *
 * @returns {int} The total number of syllables in the word.
 */
export function calculateTotalNumberOfSyllables( word ) {
	let result = 0;

	for ( let i = 0; i < word.length; i++ ) {
		if ( isVowel( word[ i ] ) ) {
			result++;
		}
	}

	return result;
}

/**
 * Stems the ending of a word based on some regexRules after checking if the word is in the exception list.
 *
 * @param {string} word         The word to stem.
 * @param {Array} regexRules    The list of regex-based rules to apply to the word in order to stem it.
 * @param {string[]} exceptions The list of words that should not get the ending removed.
 *
 * @returns {string} The stemmed word.
 */
export function removePart( word, regexRules, exceptions ) {
	if ( exceptions.includes( word ) ) {
		return word;
	}

	const removePartRegex = createRulesFromMorphologyData( regexRules );
	const withRemovedPart = buildOneFormFromRegex( word, removePartRegex );
	return withRemovedPart || word;
}

/**
 * Checks if the beginning of the word is present in an exception list.
 *
 * @param {string}   word         The word to stem.
 * @param {int}      prefixLength The length of the prefix to be trimmed before checking in the list.
 * @param {string[]} beginnings   The list of word beginnings that should be checked.
 *
 * @returns {boolean} Whether the word is found in the list with beginnings.
 */
export function checkBeginningsList( word, prefixLength, beginnings ) {
	const wordWithoutPrefix = word.slice( prefixLength );
	return beginnings.some( beginning => wordWithoutPrefix.startsWith( beginning ) );
}
