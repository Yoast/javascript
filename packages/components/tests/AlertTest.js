// External dependencies.
import React from "react";
import renderer from "react-test-renderer";

// Internal dependencies.
import Alert from "../src/Alert.js";
import { makeOutboundLink } from "@yoast/helpers";

test( "the default info Alert matches the snapshot", () => {
	const component = renderer.create(
		<Alert dismissable={ true } cookieName="infoAlert" type="info">
			Norway knighted a penguin &
			<br />
			Sweden has a rabbit show-jumping competition called Kaninhoppning.
			<p>End of <b>info</b> alert </p>
		</Alert>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the undismissable warning alert matches the snapshot", () => {
	const component = renderer.create(
		<Alert dismissable={ false } cookieName="warningAlert" type="warning">
			<i>Watch out where the huskies go,
				and don&apos;t you eat that yellow snow. </i>
			This <b>warning</b> alert cannot be dismissed.
		</Alert>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the success alert with link matches the snapshot", () => {
	const YoastLink = makeOutboundLink();
	const component = renderer.create(
		<Alert dismissable={ true } cookieName="successAlert" type="success">
			This is a <b>success</b> alert with a link in it: &nbsp;
			<YoastLink href="https://yoast.com">yoast.com </YoastLink>
		</Alert>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the error alert matches the snapshot", () => {
	const component = renderer.create(
		<Alert dismissable={ true } cookieName="errorAlert" type="error">
			This is an <b>error</b> alert! <b>Something went wrong </b> ðŸ˜¢ <br />
		</Alert>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
