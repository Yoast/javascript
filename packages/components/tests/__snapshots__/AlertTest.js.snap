// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the default info Alert matches the snapshot 1`] = `
<div
  className="cZMcrk"
>
  <span
    className="ehCebz"
  />
  <div
    className="jlXClR"
  >
    Norway knighted a penguin &
    <br />
    Sweden has a rabbit show-jumping competition called Kaninhoppning.
    <p>
      End of 
      <b>
        info
      </b>
       alert 
    </p>
  </div>
  <button
    aria-label="close"
    className="gXcXMi"
    onClick={[Function]}
  />
</div>
`;

exports[`the error alert matches the snapshot 1`] = `
<div
  className="gyOzkT"
>
  <span
    className="iLAVbb"
  />
  <div
    className="jlXClR"
  >
    This is an 
    <b>
      error
    </b>
     alert! 
    <b>
      Something went wrong 
    </b>
     ðŸ˜¢ 
    <br />
  </div>
  <button
    aria-label="close"
    className="bEZEdw"
    onClick={[Function]}
  />
</div>
`;

exports[`the success alert with link matches the snapshot 1`] = `
<div
  className="jpPwfc"
>
  <span
    className="ePcGuf"
  />
  <div
    className="jlXClR"
  >
    This is a 
    <b>
      success
    </b>
     alert with a link in it: Â 
    <a
      href="https://yoast.com"
      rel={null}
      target="_blank"
    >
      yoast.com 
      <span
        className="A11yNotice-sc-1feyioo-0 gHCoqL"
      >
        (Opens in a new browser tab)
      </span>
    </a>
  </div>
  <button
    aria-label="close"
    className="gNHmoi"
    onClick={[Function]}
  />
</div>
`;

exports[`the undismissable warning alert matches the snapshot 1`] = `
<div
  className="knYbsI"
>
  <span
    className="GBVXz"
  />
  <div
    className="jlXClR"
  >
    <i>
      Watch out where the huskies go, and don't you eat that yellow snow. 
    </i>
    This 
    <b>
      warning
    </b>
     alert cannot be dismissed.
  </div>
</div>
`;
